client/source/js/modules/model/cost-coverage-ctrl.js

```diff
@@ -254,30 +258,29 @@ define(['./module', 'underscore'], function (module, _) {
     var getPlotModel = function() {
       return {
         progname: $scope.selectedProgram.acronym,
-        ccparams: costCoverageParams(),
+        ccparams: $scope.costCoverageParams(),
         coparams: []
       };
     };
 
     /**
-     * Returns true if only some of the elements in an array are defined or not null
-     */
-    var hasSomeElements = function(params) {
-      return params.some(function(item){return item;}) && params.some(function(item){return !item;});
-    };
-
-    /**
      * Returns true if all of the elements in an array are defined or not null
      */
     var hasAllElements = function(params) {
-      return params.every(function(item){return item;});
+      return params.every(function(item) { return item; });
     };
 
     /**
-     * Returns true if none of the elements in an array is defined or not null
+     * Returns true if all of the elements in an array not defined, null or NaN
      */
-    var hasNoElements = function(params) {
-      return !params || !params.length || !hasAllElements(params);
+    var hasOnlyInvaldEntries = function(params) {
+      return params.every(function(item) {
+        return item === undefined || item === null || typeof item === "number" && isNaN(item);
+      });
+    };
+
+    $scope.areValidParams = function (params) {
```

Comment:
Why should we say that the params are valid if all entries are invalid? I don't get it.